clarity_ext/driverfile.py

```diff
@@ -63,9 +64,17 @@ def _save_file_locally(self, root):
         # The file needs to be opened in binary form to ensure that Windows line endings are used if specified
         with self.os_service.open_file(full_path, 'wb') as f:
             self.logger.debug("Writing output to {}.".format(full_path))
-            newline = self.extension.newline()
-            for line in self.specific_file_service.content():
-                f.write(line + newline)
+            # Content should be either a string or something else we can iterate over, in which case we need newline
+            content = self.specific_file_service.content()
+            if isinstance(content, basestring):
+                try:
+                    f.write(content)
+                except UnicodeEncodeError:
+                    f.write(content.encode("utf-8"))
+            else:
+                newline = self.extension.newline()
+                for line in content:
+                    f.write(line + newline)
```

Comment:
The method `content()` can return either a `basestring` (`str`/`unicode`) or a list of strings. If it's a list of string, it should use that newline attribute when writing to the file, othewise it should just write out the entire string.